.gitattributes

```diff
@@ -0,0 +1,81 @@
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto
+
+# Files with these extensions are accepted source files inside the ReactOS
+# tree. They should have native line endings of the OS.
+*.acf text
+*.asm text
+*.c text
+*.cc text
+*.cmake text
+*.cpp text
+*.cxx text
+*.h text
+*.hpp text
+*.hxx text
+*.idl text
+*.inc text
+*.inl text
+*.klc text
+*.l text
+*.lds text
+*.md text
+*.rc text
+*.s text
+*.S text
+*.sfd text
+*.spec text
+*.svg text
+*.txt text
+*.TXT text
+*.y text
+
+# Files with these extensions end up in the built ReactOS system, so they
+# need to have CRLF line endings.
+*.bat text eol=crlf
+*.cmd text eol=crlf
+*.exp text eol=crlf
+*.hhc text eol=crlf
+*.hhp text eol=crlf
+*.ini text eol=crlf
+*.INI text eol=crlf
+*.inf text eol=crlf
+*.inf.tpl text eol=crlf
+*.js text eol=crlf
+*.manifest text eol=crlf
+*.mc text eol=crlf
+*.mof text eol=crlf
+*.rgs text eol=crlf
+*.sif text eol=crlf
+*.vbs text eol=crlf
+
+# Files with these extensions are accepted binary files inside the ReactOS
+# tree. Git must not tamper with them at all!
+*.avi binary
+*.bin binary
+*.bmf binary
+*.bmp binary
+*.BMP binary
+*.chm binary
+*.cur binary
+*.dll binary
+*.emf binary
+*.gif binary
+*.ico binary
+*.jpg binary
+*.mp3 binary
+*.nls binary
+*.otf binary
+*.pdn binary
+*.pfb binary
+*.pfm binary
+*.png binary
+*.psd binary
+*.ttc binary
+*.ttf binary
+*.wav binary
+*.xcf binary
+
+# All other extensions not explicitly mentioned here are left for Git to
+# handle automatically.
```

Comment:
I seem to recall shell scripts and Makefiles don't work with CRLF line endings, so it might make sense to add those (or correct my wrong memory :p)

Reply:
depends on the shell afaik.

Reply:
We are regularly using the BuildBot shell scripts under Windows (using Cygwin).
Which are stored with svn:eol-style native in the "project-tools" repository and work fine (using SHELLOPTS=igncr :P)

Not sure about Makefiles though. It has been some time since I used them.
If we just keep them in our repo for syncing third-party code again, Git's automatic way of dealing with them should be sufficient though.

Reply:
I feel like for the average Joe not needing those SHELLOPTS would be enough of a reason to force LF on such files, but don't feel strongly enough to block a merge on this or anything.